doc/source/user/how-to-index.rst

```diff
@@ -0,0 +1,324 @@
+.. currentmodule:: numpy
+
+.. _how-to-index.rst:
+
+*****************************************
+How to index :class:`ndarrays <.ndarray>`
+*****************************************
+
+.. seealso:: :ref:`basics.indexing`
+
+This page tackles common examples. For an in-depth look into indexing, refer
+to :ref:`basics.indexing`.
+
+Access specific/arbitary rows and columns
+=========================================
+
+Use :ref:`basic-indexing` features like :ref:`slicing-and-striding`, and
+:ref:`dimensional-indexing-tools`.
+
+    >>> a = np.arange(30).reshape(2, 3, 5)
+    >>> a
+    array([[[ 0,  1,  2,  3,  4],
+            [ 5,  6,  7,  8,  9],
+            [10, 11, 12, 13, 14]],
+    <BLANKLINE>
+            [[15, 16, 17, 18, 19],
+            [20, 21, 22, 23, 24],
+            [25, 26, 27, 28, 29]]])
+    >>> a[0, 2, :]
+    array([10, 11, 12, 13, 14])
+    >>> a[0, :, 3]
+    array([ 3,  8, 13])
+    
+Note that the output from indexing operations can have different shape as the
+original object. To preserve the original dimensions after indexing, you can
+use :func:`newaxis`. To use other such tools, refer to
+:ref:`dimensional-indexing-tools`.
+
+    >>> a[0, :, 3].shape
+    (3,)
+    >>> a[0, :, 3, np.newaxis].shape
+    (3, 1)
+    >>> a[0, :, 3, np.newaxis, np.newaxis].shape
+    (3, 1, 1)
+
+Variables can also be used to index::
+
+    >>> y = 0
+    >>> a[y, :, y+3]
+    array([ 3,  8, 13])
+
+Refer to :ref:`dealing-with-variable-indices` to see how to use
+:term:`python:slice` and :py:data:`Ellipsis` in your index variables.
+
+Index along a specific axis
+---------------------------
+
+Use :meth:`take`. See also :meth:`take_along_axis` and
+:meth:`put_along_axis`.
+
+    >>> np.take(a, [2, 3], axis=2)
+    array([[[ 2,  3],
+            [ 7,  8],
+            [12, 13]],
+    <BLANKLINE>
+            [[17, 18],
+            [22, 23],
+            [27, 28]]])
+    >>> np.take(a, [2], axis=1)
+    array([[[10, 11, 12, 13, 14]],
+    <BLANKLINE>
+            [[25, 26, 27, 28, 29]]])
+
+Index columns
+-------------
+
+Use :ref:`dimensional-indexing-tools` to avoid shape mismatches::
```

Comment:
I'm not sure this is very clear on first read - maybe show what would happen if you used a "naive" approach here first?